src/Range.cpp

```diff
@@ -179,6 +179,43 @@ bool Range::encloses (const Range& other) const
   return false;
 }
 
+
+////////////////////////////////////////////////////////////////////////////////
+// This is a standard enclosure check, except that the other range
+// need not have an end time, even if this one does.
+//
+// Detect the following enclosure cases:
+//
+// this                     [--------)
+//   A                        [----)
+//   B                          [...
+//
+// this                     [...
+//   A                        [----)
+//   B                           [--------)
+//   C                                    [--------)
+//   D                         [...
+//   E                                 [...
+//
+bool Range::segmentContains (const Range& other) const
+{
+  if (is_started ())
+  {
+    if (is_ended ())
+    {
+      if (other.is_started () && other.start >= start)
+        return ! other.is_ended () || other.end <= end;
```

Comment:
At this position (line 207) we know `this` is a closed interval and the start date of `other` is within this interval. If `other` is a closed interval, `segmentContains` returns true if the end date of `other` is within `this`. If `other` is open `segmentContains` always returns true.

Why this difference between
```
this  [...]
other   [.....
```
and
```
this  [...]
other   [.....]
```
?

Reply:
`segmentContains` is used to check whether a `Range` can be stored in a `Datafile`. The range to be stored must be enclosed in the range covered by the datafile because a single range cannot extend beyond the end of a data file (e.g., `2018-01-15 - 2018-02-15` is actually stored as `2018-01-15 - 2018-01-31` in `2018-01.data` and as `2018-02-01 - 2018-02-15` in `2018-02.data`).

However, when a range doesn't have an end date, it would have to be stored in an infinite number of data files. To work around that, we instead store it in the file containing its start date. This is consistent with the current behavior.

Here, `other` is stored in the data file with range `this`:

```text
this  [...]
other   [.....
```

However, in the following case, `other` extends beyond the end of `this` and would cause an assertion failure when stored in the data file with range `this`:

```text
this  [...]
other   [.....]
```